workflow/controller/operator.go

```diff
@@ -2977,6 +2978,26 @@ func (woc *wfOperationCtx) executeSuspend(nodeName string, templateScope string,
 	return node, nil
 }
 
+func resolveInputFieldsForSuspendTemplate(tmpl *wfv1.Template) {
+	parameters := tmpl.Inputs.Parameters
+	for i, parameter := range parameters {
+		if parameter.Value != nil {
+
+			value := parameter.Value.String()
+			valueMap := map[string][]wfv1.AnyString{}
+
+			if err := json.Unmarshal([]byte(value), &valueMap); err != nil {
+				continue
+			}
+
+			if enum, ok := valueMap["enum"]; ok && len(enum) > 0 {
+				parameters[i].Enum = enum
```

Comment:
I don't think we can edit the template like this - it is GitOps unfriendly, and I'd expect it to produce weird edge-case bugs. You can change `/status/nodes`.

Reply:
Done

Reply:
Can you add the following to help me review this:

* A doc page showing how to use this (in the `docs` dir and listed in `mkdocs.yml`)
* An example in the `examples` dir.